TICC.ino

```diff
@@ -61,37 +41,23 @@ const int CALIBRATION2 = 0x1C;            // default 0x00_0000
 // max clock speed: 20 mHz
 // Using TDC7200 timing mode 1...
 
-// global vars
-long long unsigned stop_time_A;
-long long unsigned stop_time_B;
-long result_A;
-long result_B;
-int CSB;
-int ENABLE;
-
-  int timeresult_A = 0;
-  int clockresult_A = 0;
-  int calresult_A = 0;
-  int timeresult_B = 0;
-  int clockresult_B = 0;
-  int calresult_B = 0;
 
 void setup() {
   // start the SPI library:
   SPI.begin();
   
-  pinMode(int_pin, INPUT);
+  pinMode(intPin, INPUT);
   
-  pinMode(ENABLE_A,OUTPUT);
-  pinMode(INTB_A,INPUT);
-  pinMode(CSB_A,OUTPUT);
-  pinMode(STOP_A,INPUT);
-  pinMode(ENABLE_B,OUTPUT);
-  pinMode(INTB_B,INPUT);
-  pinMode(CSB_B,OUTPUT);
-  pinMode(STOP_B,INPUT);
+  pinMode(ChA->ENABLE,OUTPUT);
+  pinMode(ChA->INTB,INPUT);
+  pinMode(ChA->CSB,OUTPUT);
+  pinMode(ChA->STOP,INPUT);
+  pinMode(ChB->ENABLE,OUTPUT);
+  pinMode(ChB->INTB,INPUT);
+  pinMode(ChB->CSB,OUTPUT);
+  pinMode(ChB->STOP,INPUT);
   
-  attachInterrupt(interrupt, coarse_timer, RISING);
+  attachInterrupt(interrupt, coarseTimer, RISING);
   
   TDC_setup(ChA);
   TDC_setup(ChB);
```

Comment:
Ummm, my sources say to get rid of 'interrupt' and replace with digitalPinToInterrupt(intPin), like I sent to Tom...  At least I think that was the construct that I supplied.  'intPin' (or similar) had value 3, while 'interrupt' had value 1.

I suspect that maybe you are working off his original pull request and not what I sent him to consider?

Further, I'm thinking that possibly the TDC7200 thingie should be a class, not a struct, so that those read and write methods can operate on a class instance (an object).  You could still have an array of class instances (objects) and iterate thru it for some things.

73,
George K9TRV

-----Original Message-----
From: Jeremy McDermond [mailto:notifications@github.com] 
Sent: Wednesday, April 27, 2016 3:52 PM
To: TAPR/TICC <TICC@noreply.github.com>
Subject: Re: [TAPR/TICC] Add TICC.h file: defines hardware registers, interrupt pins, hardware… (#3)

In TICC.ino <https://github.com/TAPR/TICC/pull/3#discussion_r61324305> :

>    
> -  attachInterrupt(interrupt, coarse_timer, RISING);
> +  attachInterrupt(interrupt, coarseTimer, RISING);
>    
>    TDC_setup(ChA);
>    TDC_setup(ChB);

These just become

TDC_setup(&ChanA)
TDC_setup(&ChanB)


And it becomes clear that you're just sending pointers to the static structs used in here.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub <https://github.com/TAPR/TICC/pull/3/files/2a0b1e74f63e5e7c8fe160de49044f71416798d2#r61324305>  <https://github.com/notifications/beacon/AKfFAN3KFaYqGfa33gQWVQNGA79DfmHbks5p775dgaJpZM4IRR1w.gif>